apps/hosts/views.py

```diff
@@ -212,20 +239,63 @@ def remove_self_from_challenge_host_team(request, challenge_host_team_pk):
 @api_view(['POST'])
 @permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
 @authentication_classes((ExpiringTokenAuthentication,))
-def invite_host_to_team(request, pk):
-
+def email_invite_host_to_team(request, pk):
     try:
         challenge_host_team = ChallengeHostTeam.objects.get(pk=pk)
     except ChallengeHostTeam.DoesNotExist:
         response_data = {'error': 'ChallengeHostTeam does not exist'}
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
-    serializer = InviteHostToTeamSerializer(data=request.data,
-                                            context={'challenge_host_team': challenge_host_team,
-                                                     'request': request})
-    if serializer.is_valid():
-        serializer.save()
-        response_data = {
-            'message': 'User has been added successfully to the host team'}
+    email = request.data.get('email')
+
+    try:
+        User.objects.get(email=email)
+    except User.DoesNotExist:
+        response_data = {'error': 'User does not exist with this email address!'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    if email == request.user.email:
+        response_data = {'error': 'A host cannot invite himself'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    team_hash, email_hash = ENCODE_DATA([str(pk), email])
+    unique_hash = (team_hash + "/" + email_hash)
+    url = request.data['url'].split("team")[0]
+    full_url = url + "participant-invitation/" + unique_hash
+    team_name = challenge_host_team.team_name
+    body = "You've been invited to join the host team {}. " \
```

Comment:
Store the email body in `hosts/templates` similar to how django allauth is storing. I think I have shared the link with you earlier. Let me know if I haven't done that.

Reply:
No, I think you didn't. How do you access the template from the view?